packages/edit-site/src/components/dataviews/item-actions.js

```diff
@@ -38,27 +93,22 @@ export default function ItemActions( { item, actions } ) {
 		<HStack justify="flex-end">
 			{ !! primaryActions.length &&
 				primaryActions.map( ( action ) => (
-					<Button
-						label={ action.label }
+					<MaybeWithModal
```

Comment:
Instead of trying to use the same component for both, maybe we should just do a "switch" here on the type of action?

Reply:
In terms of primary/secondary or in terms of having a modal or not?

Reply:
having modal or not

Reply:
I updated. Is this what you had in mind?